src/lib/types/Augments.d.ts

```diff
@@ -94,6 +94,10 @@ declare module 'discord.js' {
 		emit(event: string | symbol, ...args: any[]): boolean;
 	}
 
+	interface ClientEvents {
+		[Events.TaskError]: [error: Error, payload: TaskErrorPayload];
+	}
```

Comment:
Can't this go in the block of custom emits above?

Reply:
Not without sacrificing type checking in the Event (see that they're generic).